core/lib/bfs/bucket/in_mem.rb

```diff
@@ -17,16 +17,17 @@ def initialize(encoding:, &closer)
         end
 
         def close
-          close!
-          @closer&.call(self)
+          super.tap do
+            @closer&.call(self)
+          end
         end
 
         def close!
           __getobj__.close
         end
```

Comment:
Is `.close!` still used? It was called in `.close` before, but now maybe can go?

Reply:
just to keep it consistent with the file-based one

Reply:
Maybe `.close!` should use `.close` instead of just `__getobj__.close`?

I'm a bit worried that they differ (sorry, might be getting it wrong)

Reply:
no, `#close` applies `@closer` while `#close!` just closes without the callback, same as with file-based one